dell/xps/17-9710/intel/default.nix

```diff
@@ -0,0 +1,27 @@
+{ lib, pkgs, ... }: {
+  imports = [
+    ../../../../common/cpu/intel
+    ../../../../common/pc/laptop
+    ../../../../common/pc/ssd
+  ];
+
+  # patch the kernel to enable sound
+  # Tested with 5.15.12
+  boot.kernelPatches = lib.mkDefault [{
+    name = "xps17-9710-fix-sound";
+    patch = null;
+    extraConfig = ''
+      SND_SOC_INTEL_USER_FRIENDLY_LONG_NAMES y
+      SND_SOC_INTEL_SOUNDWIRE_SOF_MACH m
+    '';
+  }];
+
+  # Includes the Wi-Fi and Bluetooth firmware
+  hardware.enableRedistributableFirmware = true;
+
+  # Requires at least 5.12 for working sound
+  boot.kernelPackages = lib.mkIf (lib.versionOlder pkgs.linux.version "5.12") (lib.mkDefault pkgs.linuxPackages_latest);
+
+  # Allows for updating firmware via `fwupdmgr`.
+  services.fwupd.enable = true;
```

Comment:
Maybe fwupd is supported by this device in the README instead and leave it to the user if they want to enable it permanently or not. Without documentation that this service exists, it wouldn't be very useful anyway.

Reply:
:+1: Sounds reasonable; I'll take this as an excuse to create readme